### PR TITLE
[CU-2vtm7m2] CI job for uploading of composable-sandbox

### PR DESCRIPTION
**Pull request by andor0**
## Issue
https://app.clickup.com/t/2vtm7m2

### CODE REVIEW

**Comment by ComposableFiB0t** [comment]
Task linked: [CU-2vtm7m2 Create CI job to upload composable-sandbox docker image](https://app.clickup.com/t/2vtm7m2)

File: .github/workflows/build-test-deploy.yml [hunk 1]
```diff
@@ -402,9 +412,10 @@ jobs:
           COMMIT_SHA=$(git rev-parse --short=9 HEAD)
           DOCKER_IMAGE_HASH=$(docker images | tail --lines=1 | tr --squeeze-repeats  ' ' |  cut --delimiter " " --fields 3)
           # TODO: make nix sha256 hash(allow to find by using nix ids) and branch dependant(allow to test from branches) tags
-          docker image tag $DOCKER_IMAGE_HASH ${{env.DOCKER_REPOSITORY_NAME}}/composable-devcontainer:$COMMIT_SHA
-          docker image tag $DOCKER_IMAGE_HASH ${{env.DOCKER_REPOSITORY_NAME}}/composable-devcontainer:latest
-          docker push --all-tags ${{env.DOCKER_REPOSITORY_NAME}}/composable-devcontainer
+          docker image tag $DOCKER_IMAGE_HASH ${{env.DOCKER_REGISTRY_NAME}}/composable-devcontainer:$COMMIT_SHA
+          docker image tag $DOCKER_IMAGE_HASH ${{env.DOCKER_REGISTRY_NAME}}/composable-devcontainer:latest
```
Comment (dzmitry-lahoda) [inline .github/workflows/build-test-deploy.yml:416]: imho retag as latest should happen only after all tests passed, and iff merge to main, like devnet

File: .github/workflows/build-test-deploy.yml [hunk 2]
```diff
@@ -340,13 +341,22 @@ jobs:
...
-          # TODO: upload docker image
-          # Only on trigger by user?
-          nix build .#devnet-container --no-update-lock-file --show-trace -L
+          # enable required features (see https://github.com/cachix/install-nix-action/issues/19)
+          mkdir -p ~/.config/nix
+          echo "system-features = kvm" >> ~/.config/nix/nix.conf
+      - run: |
+          cachix watch-exec -j 16 -c 0 composable-community nix -- build .#devnet-container --no-update-lock-file --show-trace -L
+          docker load --input ./result
+          docker tag composable-devnet-container:latest "${{ env.DOCKER_REGISTRY_NAME }}/composable-sandbox:latest"
```
Comment (dzmitry-lahoda) [inline .github/workflows/build-test-deploy.yml:358]: imho retag as latest should happen only after all tests passed, and iff merge to main, like devnet.

i mean we should not replace latest with arbitrary branch

**Comment by dzmitry-lahoda** [comment]
like this https://github.com/ComposableFi/composable/runs/8006345266?check_suite_focus=true <- latest gated only on main merge

File: .github/workflows/build-test-deploy.yml (hunk 1 above)
Comment (andor0) [reply .github/workflows/build-test-deploy.yml:416]: yes, I added `TODO` about it below.

File: .github/workflows/build-test-deploy.yml (hunk 2 above)
Comment (andor0) [reply .github/workflows/build-test-deploy.yml:358]: yes, this job will be run only from main branch (see https://github.com/ComposableFi/composable/pull/1494/files#diff-22f49dea8896855b65ec7f98c2dccafa1b7317f851952fc98ee59f29d84f8af5R312)

Comment (andor0) [reply .github/workflows/build-test-deploy.yml:358]: also, this job will be run after all tests are passed.

**Comment by andor0** [comment]
> like this https://github.com/ComposableFi/composable/runs/8006345266?check_suite_focus=true <- latest gated only on main merge

this is done, see https://github.com/ComposableFi/composable/pull/1494/files#diff-22f49dea8896855b65ec7f98c2dccafa1b7317f851952fc98ee59f29d84f8af5R312

**Review by dzmitry-lahoda — APPROVED** [review]

**Comment by dzmitry-lahoda** [comment]
@Mergifyio refresh

**Pull request merged**